### PR TITLE
:seedling: Update Kustomize version

### PR DESCRIPTION
**Pull request by peppi-lotta**
Update Kustomize version.

Also updating Kustomize in these 
- https://github.com/metal3-io/ip-address-manager/pull/506
- https://github.com/metal3-io/cluster-api-provider-metal3/pull/1625

### CODE REVIEW

File: lib/common.sh [hunk 1]
```diff
@@ -159,6 +159,9 @@ else
   export IPAMBRANCH="${IPAMBRANCH:-main}"
 fi
 
+export CAPM3BRANCH="${CAPM3BRANCH:-peppi-lotta/fix/Update-Kustomize-depricated-syntax}"
```
Comment (tuminoid) [inline lib/common.sh:162]: This kind of changes are better off put in `config_example.sh` where its clear they're not part of the intended change.

Also, mark it as WIP and add a description.

**Review by kashifest — COMMENTED** [review]
/approve

**Pull request merged**